include/onnxruntime/core/framework/allocator.h

```diff
@@ -15,21 +15,78 @@
 #include "core/framework/fence.h"
 #include "core/session/onnxruntime_c_api.h"
 
+// Struct to represent a physical device.
+struct OrtDevice {
+  typedef int8_t DeviceType;
```

Comment:
Would unsigned types work?

Reply:
both works, as they're just ids. Any strong preference here please?

Reply:
If you don't except DeviceType to be negative, you should not use signed types.